File: SumofLeftLeaves.cpp
 /*
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int ans=0;
        if(root){
            if(root->left && root->left->left==NULL && root->left->right==NULL)
                ans+=root->left->val+sumOfLeftLeaves(root->right);
            else
                ans+=sumOfLeftLeaves(root->left)+sumOfLeftLeaves(root->right);
            
        }
        
        return ans;
    }
};

File: SumofLeftLeaves.py
# definition for a binary tree node.
# class treenode:
#     def __init__(self, val=0, left=none, right=none):
#         self.val = val
#         self.left = left
#         self.right = right
class solution:
    def sumofleftleaves(self, root: optional[treenode]) -> int:
        # sum of left's leaf node
        if root==none:
            return 0
        
        stack=[(root,false)]
        ans=0
        while stack:
            p,check=stack.pop()
            
            if p.left==none and p.right==none and check:
                ans+=p.val
            if p.left:
                stack.append((p.left, true))
            if p.right:
                stack.append((p.right, false))
                
        return ans
